Make and display warning on editing an .obj mesh

ChristopheLY opened the pull request:
I displayed the button warning when the mesh .obj displayed it
It's update when you add or remove the mesh 
fixe #20659

madmiraal committed:
Enable Bullet DEBUG on debug builds.

Calinou committed:
Rename "Identifier" to "Bundle Identifier" in macOS/iOS export presets

"Bundle Identifier" is more well-understood among macOS and iOS
developers and is less ambiguous.

This is a slight breaking change as export presets will need to be
updated to account for this change.

See https://github.com/godotengine/godot-docs/pull/3295.

RandomShaper committed:
Fix potential crash when listing leaked objects

Note:
Casting to the C++ classes and calling the methods there would work as well,
but would require including he header files for the specific object types handled
here, which wouldn't be OK either.

ChristopheLY committed:
editor help doc sort with space for Methods and Properties by alphabetical order

mrushyendra committed:
Enable zero padding with float specifier for format strings

Godot currently supports zero padding for integers, octals and
hexadecimals when using format strings, but not for floats.

This commit adds support for zero padding for floats, thus ensuring
consistent behavior for all types, and making Godot's format specifiers'
behavior closer to c's `printf()`.

Before: `print("<%07.2f>" % -0.2345)` prints `<  -0.23>`.
Now: `print("<%07.2f>" % -0.2345)` prints `<-000.23>`.
     `print("<%7.2f>" % -0.2345)` prints `<  -0.23>`.

ThakeeNathees committed:
shodowed var warning for `for` loop counter

Fix: #39268

ThakeeNathees committed:
predefined var check for `for` loop counter

mbrlabs committed:
Re-apply basis of newly added node (via drag-n-drop) in editor viewport

akien-mga committed:
GDScript LSP: Fix crash in notify_client

`latest_client_id` now defaults to `-1` (invalid ID) instead of `0`.

Also fix typo in notification `gdscrip_client/changeWorkspace`,
and fix argument names in method binds.

Fixes #39375.

akien-mga committed:
Core: Add hints to run with --verbose when leaking nodes/resources at exit

madmiraal committed:
Test collision mask before creating constraint pair in Godot physics broadphase 2D and 3D.

Commit:
Renamed InputMap.get_action_list to InputMap.get_action_events

mrushyendra committed:
Account for file deletion and renaming in Export Presets

Ensure that presets are updated with the latest files when
starting up or opening the Project Export dialog. Fixes the
error where Godot would attempt to export deleted files that
were previously selected.

bruvzg committed:
macOS export, add notarization support and id / signing / notarization config check.

akien-mga committed:
i18n: Sync translations with Weblate

(cherry picked from commit d68e631566e953e09e71438558b9f1376648d025)

bruvzg committed:
Add keyboard layout enumeration / set / get functions (macOS, Windows, Linux/X11), remove latin variant function.

akien-mga committed:
String: Use ABS macro in padding code

Follow-up to #39261.

mrushyendra committed:
Fix whole word search slowdown in editor

Reduce repeated iteration through the full text
when counting the number of occurrences of whole
words while searching a file in the editor.

madmiraal committed:
Ensure pair callback data is set to null when it's null.

mrushyendra committed:
Fix match count for whole word search in editor

Check if a match borders a new line char when incrementing match counts.

akien-mga committed:
GDScript LSP: Fix wrong error checks added in #39385

Reverts `latest_client_id` back to 0, as I misunderstood how the client
IDs are assigned and, without further testing and debugging, I can't
say if this was a bug or a valid default value.
Similarly, a `latest_client_id` of -1 is no longer raising an error.

Fixes #39548.

akien-mga committed:
i18n: Sync translations with Weblate

Remove 'de_CH' translation (4% complete) to default to the 100%
complete 'de' one.

Fixes #32625.

(cherry picked from commit 35a5ebb44d534683da4e2e6b774fe7e1d165a24b)

Calinou committed:
Document Mono-specific classes

This fully documents all Mono-specific classes.

ChristopheLY committed:
make and display warning on editing an .obj mesh, fixes #20659

akien-mga committed:
DocData: Skip language-specific ClassDoc without methods/constants

Removes the useless `@C#`, `@NativeScript` and `@VisualScript` entries.

groud commented on editor/editor_node.cpp:
```diff
@@ -1789,6 +1794,46 @@ void EditorNode::_edit_current() {
 			if (FileAccess::exists(source_scene + ".import")) {
 				editable_warning = TTR("This scene was imported, so changes to it won't be kept.\nInstancing it or inheriting will allow making changes to it.\nPlease read the documentation relevant to importing scenes to better understand this workflow.");
 			}
+			else if (current_node->is_class("MeshInstance3D")){
```
I don't think is the right place to do so. Usually, we put that kind of code in their respective EditorPlugin, definitely not in the editor_node.cpp file.

akien-mga commented:
This needs to be rebased to remove all the merge commits and changes which shouldn't be in this PR. It should only have one commit handle the .obj mesh warning.

See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) for instructions.

akien-mga committed:
doc: Fix rebase error from #39589

mrushyendra committed:
Fix overwrite of manual changes in export_presets.cfg when export dialog is opened

Ensures that behavior when export_presets.cfg file is edited manually
while editor is running remains identical to before commit 44094b0.

akien-mga committed:
i18n: Sync translations with Weblate

(cherry picked from commit 000db65d40ede5b8cab1ce91f604c359e66f10bb)

timoschwarzer committed:
Remove GodotPayment android plugin

This is now available in a separate repository
at https://github.com/godotengine/godot-google-play-billing

ChristopheLY committed:
ok
